common/xdc/ndc_history_resender.go

```diff
@@ -177,6 +190,30 @@ func (n *NDCHistoryResenderImpl) SendSingleWorkflowHistory(
 	return nil
 }
 
+func (n *NDCHistoryResenderImpl) GetSingleWorkflowHistoryPagingIterator(
+	ctx context.Context,
+	remoteClusterName string,
+	namespaceID namespace.ID,
+	workflowID string,
+	runID string,
+	startEventID int64,
+	startEventVersion int64,
+	endEventID int64,
+	endEventVersion int64,
+) collection.Iterator[historyBatch] {
+	return collection.NewPagingIterator(n.getPaginationFn(
+		ctx,
+		remoteClusterName,
+		namespaceID,
+		workflowID,
+		runID,
+		startEventID,
+		startEventVersion,
+		endEventID,
+		endEventVersion,
+	))
+}
```

Comment:
after this PR is done, you should 
1. move the logic in this file to history/replication package
2. support batch resend workflow tasks